src/messages.rs

```diff
@@ -194,6 +194,11 @@ pub struct GroupList {
 }
 
 impl GroupList {
+    /// Gets the `route`-th name in the group list.
+    pub fn get_routeth_name(&self, dst_name: &XorName, route: usize) -> &XorName {
+        RoutingTable::get_routeth_name(self.pub_ids.iter().map(|id| id.name()), dst_name, route)
+    }
```

Comment:
Not really - we still use `RT:get_routeth_name` in `RT::get_routeth_node`, which is in turn used in `RT::get_targets`...

And since we're sorting by names, if we still want to have a single implementation of getting the route-th something, we'd probably have to use `Into<T>` in the function signature in RT if we wanted to pass the public ids directly (and implement it for `PublicId`).